pkg/build/builder/docker.go

```diff
@@ -210,6 +214,11 @@ func (d *DockerBuilder) addBuildParameters(dir string) error {
 	instructions := dockerfile.ParseTreeToDockerfile(node)
 
 	// Overwrite the Dockerfile.
+	f, err := os.Open(dockerfilePath)
```

Comment:
Sorry I realized, you want to use the same mode that was already in the file. For that you can do os.Stat() and the fileInfo struct returned has a Mode() method.